src/charm.py

```diff
@@ -571,6 +603,104 @@ def _gnb_subnet_config_is_valid(self) -> bool:
     def _get_gnb_subnet_config(self) -> Optional[str]:
         return self.model.config.get("gnb-subnet")
 
+    def _get_access_nad_config(self) -> Dict[Any, Any]:
+        """Get access interface NAD config.
+
+        Returns:
+            config   (dict): Access interface NAD config
+
+        """
+        config = {
+            "cniVersion": "0.3.1",
+            "ipam": {
+                "type": "static",
+                "routes": [
+                    {
+                        "dst": self._get_gnb_subnet_config(),
+                        "gw": self._get_access_network_gateway_ip_config(),
+                    },
+                ],
+                "addresses": [
+                    {
+                        "address": self._get_access_network_ip_config(),
+                    }
+                ],
+            },
+            "capabilities": {"mac": True},
+        }
+        if self._get_access_interface_mtu_config():
+            config.update({"mtu": self._get_access_interface_mtu_config()})  # type: ignore
```

Comment:
I understand that  the _get_access_nad_config method is not executed if the access interface mtu config is invalid. I am just worried that someone in the future wants to reuse this methods and find themselves with an invalid valid of the access interface mtu config because we don't check whether the value is valid

Reply:
We are not validating the config settings in _get_acccess_nad_config method. But, this is valid for all config settings used in this method (gnb-subnet, access-interface-gw, etc) and not only for access-interface-mtu.  I found it useful to validate any config before constituting the NAD configs.  If any config is invalid, the execution should not come to NAD creation. Validating manual user inputs as much as earlier seems more beneficial. I think it will not be a problem to reuse these methods and developer could easily discover it.